src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -103,46 +100,25 @@ public DeprovisionWorkflowTransportAction(
 
     @Override
     protected void doExecute(Task task, WorkflowRequest request, ActionListener<WorkflowResponse> listener) {
-        // Retrieve use case template from global context
         String workflowId = request.getWorkflowId();
-        GetRequest getRequest = new GetRequest(GLOBAL_CONTEXT_INDEX, workflowId);
+        GetWorkflowStateRequest getStateRequest = new GetWorkflowStateRequest(workflowId, true);
 
         // Stash thread context to interact with system index
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            client.get(getRequest, ActionListener.wrap(response -> {
+            client.execute(GetWorkflowStateAction.INSTANCE, getStateRequest, ActionListener.wrap(response -> {
                 context.restore();
+                // Get a map of step id to created resources
+                final Map<String, ResourceCreated> resourceMap = response.getWorkflowState()
+                    .resourcesCreated()
+                    .stream()
+                    .collect(Collectors.toMap(ResourceCreated::workflowStepId, Function.identity()));
```

Comment:
Preserve the order here:
```suggestion
                    .collect(Collectors.toMap(ResourceCreated::workflowStepId, Function.identity(), (u, m) -> u, LinkedHashMap::new));
```